pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -75,16 +76,25 @@ func Add(mgr manager.Manager) error {
 }
 
 func (s *secretResource) Process() (reconcile.Result, error) {
-	secretExists, err := s.secretExists(s.infinispan.GetAdminSecretName())
+	secret, err := s.getSecret(s.infinispan.GetAdminSecretName())
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	// If the operator secret does not already exist create it with generated identities
-	if !secretExists {
+	if secret == nil {
 		if err = s.createAdminIdentitiesSecret(); err != nil {
 			return reconcile.Result{}, err
 		}
+	} else {
+		// Patch secret to include any generated fields
+		updated, err := s.update(secret, func() error {
+			return s.addCliProperties(secret)
```

Comment:
@dmvolod I was envisaging that you could add an additional call to this update function and in `createSecret()` to a method that generates the monitoring `username` & `password` fields

Reply:
Yes, cool, @ryanemerson 
We need to merge this PR before and I think this section will be permanent and never been deleted